cli/dcoscli/main.py

```diff
@@ -67,26 +63,28 @@ def _main():
             msg = 'Unable to get the cluster_id of the cluster.'
             logger.exception(msg)
 
-    # the call to retrieve cluster_id must happen before we run the subcommand
-    # so that if you have auth enabled we don't ask for user/pass multiple
-    # times (with the text being out of order) before we can cache the auth
-    # token
-    subproc = Popen([executable,  command] + args['<args>'],
-                    stderr=PIPE)
+    # send args call to segment.io
+    with ThreadPoolExecutor(max_workers=2) as reporting_executor:
+        analytics.segment_track_cli(reporting_executor, config, cluster_id)
 
-    if dcoscli.version != 'SNAPSHOT':
-        return analytics.wait_and_track(subproc, cluster_id)
-    else:
-        return analytics.wait_and_capture(subproc)[0]
+        # the call to retrieve cluster_id must happen before we run the
+        # subcommand so that if you have auth enabled we don't ask for
+        # user/pass multiple times (with the text being out of order)
+        # before we can cache the auth token
+        if command in subcommand.default_subcommands():
+            sc = subcommand.SubcommandMain(command, args['<args>'])
+        else:
+            executable = subcommand.command_executables(command)
+            sc = subcommand.SubcommandProcess(
+                executable, command, args['<args>'])
 
+        exitcode, err = sc.run_and_capture()
 
-def _doc():
-    """
-    :rtype: str
-    """
-    return pkg_resources.resource_string(
-        'dcoscli',
-        'data/help/dcos.txt').decode('utf-8')
+        if err:
+            analytics.track_err(
+                reporting_executor, exitcode, err, config, cluster_id)
+
+    return exitcode
```

Comment:
Move this to inside the `with`. Python is fancy but technically `exitcode` is only define inside the `with`.